ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -414,7 +414,7 @@ protected CompletableFuture<Long> appendEntryImpl(LogEntryProto entry) {
       } else {
         cache.appendEntry(entry, LogSegment.Op.WRITE_CACHE_WITHOUT_STATE_MACHINE_CACHE);
       }
-      return writeFuture;
+      return writeFuture.whenComplete((clientReply, exception) -> appendEntryTimerContext.stop());
```

Comment:
@szetszwo shall we stop timer sync or async?

This looks like my initial solution for RATIS-1810:
https://github.com/apache/ratis/compare/da274e6f92bc56dfa7acebdce186ddc05acf6352

Reply:
@kaijchen I think we can notify the waiter when writeFuture completes, instead of waiting for writeFuture and stopTimer both to be completed. However, given that stopTimer is a trivial task, it may not have significant differences.

Reply:
I'm OK with both. I was wondering which style is preferred.
Just a week ago, we chosed the async one here: https://github.com/apache/ratis/pull/847#pullrequestreview-1331830501

Reply:
@kaijchen , you are right that we should stop the timer asynchronously. i.e.
```java
      writeFuture.whenComplete((clientReply, exception) -> appendEntryTimerContext.stop());
      return writeFuture;
```

@BUAAserein , could you make the change?

Reply:
Done. I just learned the difference here. Thanks for your discussion.